airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/keys_to_lower_transformation.py

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import InitVar, dataclass, field
+from typing import Any, List, Mapping, Optional, Type, Union, Dict
+
+import dpath
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.types import Config, FieldPointer, Record, StreamSlice, StreamState
+
+
+@dataclass
+class KeysToLowerTransformation(RecordTransformation):
+
+    parameters: InitVar[Mapping[str, Any]]
+
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        pass  # parameters does not need to be considered here
+
+    def transform(
+        self,
+        record: Dict[str, Any],
+        config: Optional[Config] = None,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+    ) -> None:
+        for key in set(record.keys()):
+            record[key.lower()] = record.pop(key)
+        return None  # type: ignore  # the typing from the parent class is inaccurate based on its usage
```

Comment:
do we even need to `return None`?

Reply:
This comment was before the typing change so I'll remove that. Thanks for pointing that out